src/components/Button/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import { Link } from './index.styles';
+
+interface ButtonProps {
+  link: string;
+  text: string;
+  background?: string;
+  size?: string;
+  target?: string;
+}
+
+export default class Button extends Component<ButtonProps> {
+  render() {
+    const { link, background, size, text, target } = this.props;
+
+    // check if link is external or not
+    if (link.includes('https')) {
+      // return external
+      return (
+        <Link
+          href={link}
+          background={background}
+          size={size}
+          target={target === '_blank' ? '_blank' : '_self'}
+        >
+          {text}
```

Comment:
Should we consider using the default `{children}` from react so that this:
```javascript
      <Button
        link="https://thecodingtrain.com/"
        text="subscribe on youtube"
        background="red"
        size="fat"
      />
```
Would become:
```javascript
      <Button
        link="https://thecodingtrain.com/"
        background="red"
        size="fat"
      >
        subscribe on youtube
      </Button>
```

Not sure what the pros/cons are?